src/components/buttons/BilingualButton.tsx

```diff
@@ -0,0 +1,184 @@
+import Image from 'next/image';
+import * as React from 'react';
+import { GoTriangleDown, GoTriangleUp } from 'react-icons/go';
+
+import clsxm from '@/lib/clsxm';
+
+import enFlag from '@/icons/enflag.svg';
+import vnFlag from '@/icons/vnflag.svg';
+
+const ButtonVariant = ['default', 'light', 'dark'] as const;
+const ButtonSize = ['base', 'large', 'small'] as const;
+
+type ButtonProps = {
+  variant?: (typeof ButtonVariant)[number];
+  size?: (typeof ButtonSize)[number];
+  rightIcon?: boolean;
+  textColor?: string;
+  open: boolean;
+} & React.ComponentPropsWithRef<'button'>;
+
+type LanguageType = {
+  name: Language;
+  flag: string;
+};
+
+type VariantType = {
+  default: string[];
+  light: string[];
+  dark: string[];
+};
+
+type Language = 'English' | 'Tiếng Việt';
+
+const VariantArr: VariantType = {
+  default: [
+    'bg-white',
+    'hover:bg-[#ecf0f1]',
+    'active:bg-[#bdc3c7]',
+    'disabled:bg-white',
+  ],
+
+  light: [
+    'bg-white',
+    'hover:bg-[#ecf0f1]',
+    'active:bg-[#bdc3c7]',
+    'disabled:bg-white',
+  ],
+  dark: [
+    'bg-[#576574]',
+    'hover:bg-[#bdc3c7]',
+    'active:bg-[#7f8c8d]',
+    'disabled:bg-white',
+  ],
+};
+
+const LanguageArr: LanguageType[] = [
+  {
+    name: 'English',
+    flag: enFlag,
+  },
+  {
+    name: 'Tiếng Việt',
+    flag: vnFlag,
+  },
+];
+
+const BilingualButton = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children, className, variant, size, rightIcon, open, textColor }, ref) => {
```

Comment:
this should receive an array of options as props. Make it a shared component also pls